src/main.cpp

```diff
@@ -13,10 +13,17 @@
 #include <utility>
 
 #include "xeus/xkernel.hpp"
-#include "xeus/xkernel_configuration.hpp"
+#if defined(max)
```

Comment:
Well it seems to be that zmq includes Windows headers, which somehow define `min` and `max` macros. Consequently the build breaks...
In my view platform headers should never be included in public crossplatform headers. The Windows headers can easily break the build...

Reply:
I don't think that this should be required. We can carefully wrap calls to std::min and std::max with parentheses to avoid macro expansion.

Reply:
Old PR, but the min/max issue can be solved by defining NOMINMAX before the Windows.h include. It is usually used along with WIN32_LEAN_AND_MEAN or WIN32_EXTRA_LEAN, that remove extra garbage.